pkg/osquery/tables/windowsupdate/windowsupdate.go

```diff
@@ -0,0 +1,184 @@
+	HistoryTable
+)
+
+type Table struct {
+	client    *osquery.ExtensionManagerClient
+	logger    log.Logger
+	queryFunc queryFuncType
+	name      string
+}
+
+func TablePlugin(mode tableMode, client *osquery.ExtensionManagerClient, logger log.Logger) *table.Plugin {
+
+	columns := dataflattentable.Columns(
+		table.TextColumn("locale"),
+		table.IntegerColumn("is_default"),
+	)
+
+	t := &Table{
+		client: client,
+		logger: logger,
+	}
+
+	switch mode {
+	case UpdatesTable:
+		t.queryFunc = queryUpdates
+		t.name = "kolide_windows_updates"
+	case HistoryTable:
+		t.queryFunc = queryHistory
+		t.name = "kolide_windows_update_history"
+	}
+
+	return table.NewPlugin(t.name, columns, t.generate)
+}
+
+func queryUpdates(searcher *windowsupdate.IUpdateSearcher) (interface{}, error) {
+	return searcher.Search("Type='Software'")
+}
+
+func queryHistory(searcher *windowsupdate.IUpdateSearcher) (interface{}, error) {
+	return searcher.QueryHistoryAll()
+}
+
+type queryFuncType func(*windowsupdate.IUpdateSearcher) (interface{}, error)
+
+func (t *Table) generate(ctx context.Context, queryContext table.QueryContext) ([]map[string]string, error) {
+	var results []map[string]string
+
+	for _, locale := range tablehelpers.GetConstraints(queryContext, "locale", tablehelpers.WithDefaults("_default")) {
+		result, err := t.searchLocale(locale, queryContext)
+		if err != nil {
+			level.Info(t.logger).Log("msg", "got error searching", "locale", locale, "err", err)
+			continue
+		}
+		results = append(results, result...)
+
+	}
+
+	return results, nil
+
+}
+
+func (t *Table) searchLocale(locale string, queryContext table.QueryContext) ([]map[string]string, error) {
+	comshim.Add(1)
+	defer comshim.Done()
+
+	var results []map[string]string
+
+	searcher, setLocale, isDefaultLocale, err := getSearcher(locale)
+	if err != nil {
+		return nil, errors.Wrap(err, "new searcher")
+	}
+
+	searchResults, err := t.queryFunc(searcher)
+	if err != nil {
+		return nil, errors.Wrap(err, "search")
+	}
+
+	for _, dataQuery := range tablehelpers.GetConstraints(queryContext, "query", tablehelpers.WithDefaults("*")) {
+		flatData, err := t.flattenOutput(dataQuery, searchResults)
+		if err != nil {
+			level.Info(t.logger).Log("msg", "flatten failed", "err", err)
+			continue
+		}
+
+		rowData := map[string]string{
+			"locale":     setLocale,
+			"is_default": strconv.Itoa(isDefaultLocale),
+		}
+
+		results = append(results, dataflattentable.ToMap(flatData, dataQuery, rowData)...)
+	}
+
+	return results, nil
+}
+
+func (t *Table) flattenOutput(dataQuery string, searchResults interface{}) ([]dataflatten.Row, error) {
+	flattenOpts := []dataflatten.FlattenOpts{}
+
+	if dataQuery != "" {
+		flattenOpts = append(flattenOpts, dataflatten.WithQuery(strings.Split(dataQuery, "/")))
+	}
+
+	if t.logger != nil {
+		flattenOpts = append(flattenOpts,
+			dataflatten.WithLogger(level.NewFilter(t.logger, level.AllowInfo())),
+		)
+	}
+
+	// Works better if we bounce through json. yuck
```

Comment:
Works better in what way?

 If it's just for ease of dataflattening (i.e. not needing to implement a new dataflatten routine for the `ISearchResult` etc types), that seems very reasonable to me.

Reply:
Correct. I've updated the comment.